### ISSUE TITLE
Could pkgstats parse dependencies?

### ISSUE THREAD

**Issue by maelle** [ropensci-review-tools/pkgstats#21]
I.e. each column would be not a string but instead a list to make it easier to retrieve the vector of deps.

**Comment by maelle** [comment]
also when there is no dependency of a given type could it be a NULL instead of a NA?

**Comment by mpadge** [comment]
Short answer: No and no. These things have to be returned as single character strings to (easily) enable data from multiple packages to be combined with simple `rbind`. That *could* be done with `I(as.list(...))`, but (1) gets messy; and (2) with 120,000 rows and now > 90 columns, the increase in memory footprint would be considerable. For the same reason of `rbind`-ability, no values can be empty (`NULL` or `character(0)`), but have to have non-zero size, hence `NA` is necessary. Does that make sense?

**Comment by maelle** [comment]
yep, so this parsing should happen in pkgcheck, correct?

**Comment by mpadge** [comment]
Yes, correct. Another reason they're like that is because it's the direct output of `data.frame(read.dcf("DESCRIPTION"))$Imports)`, and `pkgstats` tries to get information out as efficiently as possible, with no unnecessary post-processing.  This is the kind of price such post-processing has to pay:

``` r
bench::mark (
             read.dcf("DESCRIPTION")[1,"Imports"],
             data.frame(read.dcf("DESCRIPTION"))$Imports,
             desc::desc_get_deps("."),
             check = FALSE)
#> # A tibble: 3 × 6
#>   expression                                       min   median `itr/sec`
#>   <bch:expr>                                  <bch:tm> <bch:tm>     <dbl>
#> 1 read.dcf("DESCRIPTION")[1, "Imports"]       245.75µs 272.87µs     3513.
#> 2 data.frame(read.dcf("DESCRIPTION"))$Imports 365.14µs 402.42µs     2269.
#> 3 desc::desc_get_deps(".")                      5.43ms   6.78ms      134.
#> # … with 2 more variables: mem_alloc <bch:byt>, gc/sec <dbl>
```

<sup>Created on 2021-09-24 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1.9000)</sup>

Extracting in current form is over 1,000 times more efficient!
